crates/solidity/outputs/cargo/crate/src/backend/passes/p1_flatten_contracts.rs

```diff
@@ -61,6 +61,8 @@ impl Transformer for Pass {
             .unwrap_or_default();
         let storage_layout = source.specifiers.iter().find_map(|specifier| {
             if let input::ContractSpecifier::StorageLayoutSpecifier(storage_layout) = specifier {
+                // TODO(validation): check that the expression is not binding constant variables up until 0.8.31
+                // https://www.soliditylang.org/blog/2025/12/03/solidity-0.8.31-release-announcement
```

Comment:
It's probably better to either add this validation comment, or if you want to disable binding for <0.8.31 in the resolution visitor [here](https://github.com/manastech/slang/blob/17fe5944b21bf904db69a3c67b79577ff4f4dc70/crates/solidity/outputs/cargo/crate/src/backend/passes/p5_resolve_references/visitor.rs#L34-L36).

Reply:
Makes sense, added the comment for now to keep this PR to the bare minimum